src/server.c

```diff
@@ -5452,31 +5551,32 @@ sds genRedisInfoString(const char *section) {
     /* Get info from modules.
      * if user asked for "everything" or "modules", or a specific section
      * that's not found yet. */
-    if (everything || modules ||
-        (!allsections && !defsections && sections==0)) {
+    if (!server.sentinel_mode && (everything || modules || (!all_sections && sections==0))) {
```

Comment:
i think we should go here on `INFO everything`, `INFO modules`, and any case where a requested section was not found (i.e. the number of emitted sections is lower than the number of requested sections)
```suggestion
    if (everything || dictFind(section_dict,"modules") != NULL || sections < dictSize(sections_dict)) {
```

Reply:
Addressed